aicsimageio/writers/ome_tiff_writer.py

```diff
@@ -0,0 +1,641 @@
+        image_name: Union[str, List[Union[str, None]], None] = None,
+        pixels_physical_size: Union[
+            Tuple[float, float, float], List[Tuple[float, float, float]], None
+        ] = None,
+        channel_colors: Union[List[int], List[List[int]], None] = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Write a data array to a file.
+
+        Parameters
+        ----------
+        data: Union[List[types.ArrayLike], types.ArrayLike]
+            The array of data to store. Data arrays must have 2 to 6 dimensions.  If a
+            list is provided, then it is understood to be multiple images written to the
+            ome-tiff file. All following metadata parameters will be expanded to the
+            length of this list.
+        uri: types.PathLike
+            The URI or local path for where to save the data.
+        dim_order: Union[str, list[str], None]
+            The dimension order of the provided data.
+            Dimensions must be a list of T,C,Z,Y,Z,S (S=samples for rgb data).
+            Dimension strings must be same length as number of dimensions in the data.
+            If S is present it must be last and its data count must be 3 or 4.
+            Default: None.
+            If None is provided for any data array, we will guess dimensions based on a
+            TCZYX ordering.
+            In the None case, data will be assumed to be scalar, not RGB.
+        ome_xml: Union[str, OME, None]
+            Provided OME metadata. The metadata can be an xml string or an OME object
+            from ome-types.  A provided ome_xml will override any other provided
+            metadata arguments.
+            Default: None
+            The passed-in metadata will be validated against current OME_XML schema and
+            raise exception if invalid.
+            The ome_xml will also be compared against the dimensions of the input data.
+            If None is given, then OME-XML metadata will be generated from the data
+            array and any of the following metadata arguments.
+        channel_names: Optional[Union[List[str], List[Optional[List[str]]]]]
+            Lists of strings representing the names of the data channels
+            Default: None
+            If None is given, the list will be generated as a 0-indexed list of strings
+            of the form "Channel:image_index:channel_index"
+        image_names: Union[str, List[str], None]
+            List of strings representing the names of the images
+            Default: None
+            If None is given, the list will be generated as a 0-indexed list of strings
+            of the form "Image:image_index"
+        pixels_physical_size: Union[Tuple[float, float, float],
+                List[Tuple[float, float, float]], None]
+            List of numbers representing the physical pixel sizes in x,y,z in microns
+            Default: None
+            If None is given, pixel size will be (1.0, 1.0, 1.0) for all images
```

Comment:
Add can this parameter be name changed from `pixels_physical_size` to `physical_pixels_sizes` to keep consistency?

Reply:
Oh yeah, I meant to do that. Will do.

Reply:
> Can you swap this to Z, Y, X as the rest of the library follows this order. And allow `PhysicalPixelSizes` the object returned from `AICSImage` to be passed in?

How do you feel about just accepting types.PhysicalPixelSizes INSTEAD of Tuple[float,float,float] ?

Reply:
I like it.